body/index.ts

```diff
@@ -0,0 +1,102 @@
+import { Readable } from "stream";
+
+export type GetBodyOpts = {
+  encoding: string;
+  limit?: number;
+  expectedLength?: number;
+};
+
+export function getBody(stream: Readable, opts: GetBodyOpts): Promise<string> {
+  return new Promise((resolve, reject) => {
+    getBodySync(stream, opts, (err?: Error, buffer?: string) => {
```

Comment:
Do we really need to get it sync inside of this async function? I feel like that might just be a shortcut they used but I'm not done reading the implementation.